src/OrchardCore.Modules/OrchardCore.Workflows/Services/SecurityTokenService.cs

```diff
@@ -2,41 +2,63 @@
 using Microsoft.AspNetCore.DataProtection;
 using Microsoft.Extensions.Localization;
 using Newtonsoft.Json;
+using OrchardCore.Modules;
 
 namespace OrchardCore.Workflows.Services
 {
     public class SecurityTokenService : ISecurityTokenService
     {
-        private readonly Lazy<IDataProtector> _dataProtector;
+        private readonly IDataProtector _dataProtector;
+        private readonly IClock _clock;
 
-        public SecurityTokenService(IDataProtectionProvider dataProtectionProvider, IStringLocalizer<SecurityTokenService> localizer)
+        public SecurityTokenService(
+            IDataProtectionProvider dataProtectionProvider, 
+            IClock clock,
+            IStringLocalizer<SecurityTokenService> localizer)
         {
-            _dataProtector = new Lazy<IDataProtector>(() => dataProtectionProvider.CreateProtector("OrchardCore.Workflows.Services.SecurityTokenService.V1"));
+            _dataProtector = dataProtectionProvider.CreateProtector("Tokens");
+            _clock = clock;
             T = localizer;
         }
 
         private IStringLocalizer T { get; }
 
-        public string CreateToken<T>(T payload)
+        public string CreateToken<T>(T payload, TimeSpan lifetime)
         {
-            var json = JsonConvert.SerializeObject(payload);
-            var token = _dataProtector.Value.Protect(json);
+            var expiringPayload = new ExpiringPayload<T> { Payload = payload, ExpireUtc = _clock.UtcNow.Add(lifetime) };
```

Comment:
You can muchly simplify that file using `ITimeLimitedDataProtector`.

Reply:
https://github.com/aspnet/DataProtection/blob/dev/src/Microsoft.AspNetCore.DataProtection.Extensions/DataProtectionAdvancedExtensions.cs#L89